### PR TITLE
feat: fund client ng from devimint dev-fed

### LINKED ISSUE #2507: `devimint dev-fed` should fund new cli client
Probably best to wait for https://github.com/fedimint/fedimint/pull/2416 and just do a peg-in

### PR DESCRIPTION
**Pull request by jp1ac4**
This is to resolve #2507.

I've created a new method `pegin_ng` based on the changes from #2416. This is called as part of `devimint dev-fed` to fund the new client and also as part of the CLI tests where I took the original code from.

I've updated some CLI tests to reflect the non-zero initial balance of the new client.

I've also reverted the changes from #2509.

### CODE REVIEW

File: devimint/src/main.rs [hunk 1]
```diff
@@ -553,13 +553,14 @@ async fn cli_tests(dev_fed: DevFed) -> Result<()> {
     info!("Testing Client-NG");
     // ## reissue e-cash
     info!("Testing reissuing e-cash");
+    const CLIENT_NG_INITIAL_BALANCE: u64 = 10_000_000; // from initial pegin
```
Comment (Maan2003) [inline devimint/src/main.rs:556]: tests shouldn't have to deal with auto pegin.

should do pegin in dev-fed subcommand handler, not in `dev_fed` function

Comment (jp1ac4) [reply devimint/src/main.rs:556]: OK that makes sense thanks.

I made a naive attempt as follows but then it breaks the old client pegin when running `just mprocs`:
```diff
@@ -1176,8 +1165,9 @@ async fn main() -> Result<()> {
         }
         Cmd::DevFed => {
             let (process_mgr, task_group) = setup(args.common).await?;
-            let _daemons =
+            let daemons =
                 write_ready_file(&process_mgr.globals, dev_fed(&process_mgr).await).await?;
+            daemons.fed.pegin_ng(10_000).await?;
             task_group.make_handle().make_shutdown_rx().await.await?;
         }

```
I'll be travelling a bit for the next few days so will try to resolve soon.

Comment (Maan2003) [reply devimint/src/main.rs:556]: do pegin before calling write_ready_file

Comment (jp1ac4) [reply devimint/src/main.rs]: Ah yes thanks, that fixed it :)

**Comment by jp1ac4** [comment]
I've moved the pegin to the `DevFed` subcommand handler (and squashed and rebased commits).

This is ready for review again.

**Review by Maan2003 — APPROVED** [review]
Thanks!
Would be nice to move the rest of mprocs startup to devimint.

**Pull request merged**